test/1_stdlib/Optional.swift

```diff
@@ -16,211 +21,133 @@ extension ImplicitlyUnwrappedOptional where Wrapped : TestProtocol1 {
-  print(c0 == nil) // DISABLED-CHECK-NEXT: true
-  print(c0 != nil) // DISABLED-CHECK-NEXT: false
-
-  print(nil == c1) // DISABLED-CHECK-NEXT: false
-  print(nil != c1) // DISABLED-CHECK-NEXT: true
-  print(nil == c0) // DISABLED-CHECK-NEXT: true
-  print(nil != c0) // DISABLED-CHECK-NEXT: false
-  */
-  
+
+  expectFalse(v1 == nil)
+  expectTrue(v1 != nil)
+  expectTrue(v0 == nil)
+  expectFalse(v0 != nil)
+
+  expectFalse(nil == v1)
+  expectTrue(nil != v1)
+  expectTrue(nil == v0)
+  expectFalse(nil != v0)
+
   let e0: E? = nil
   let e1: E? = E()
   
-  print(e1 == nil) // CHECK-NEXT: false
-  print(e1 != nil) // CHECK-NEXT: true
-  print(e0 == nil) // CHECK-NEXT: true
-  print(e0 != nil) // CHECK-NEXT: false
-
-  print(nil == e1) // CHECK-NEXT: false
-  print(nil != e1) // CHECK-NEXT: true
-  print(nil == e0) // CHECK-NEXT: true
-  print(nil != e0) // CHECK-NEXT: false
-}
-nilComparison()
+  expectFalse(e1 == nil)
+  expectTrue(e1 != nil)
+  expectTrue(e0 == nil)
+  expectFalse(e0 != nil)
 
-var counter = 0
-func nextCounter() -> Int {
-  return counter++
-}
+  expectFalse(nil == e1)
+  expectTrue(nil != e1)
+  expectTrue(nil == e0)
+  expectFalse(nil != e0)
 
-let a: Int? = 123
-let b: Int? = nil
-let c: Int? = nil
-let d: Int? = 456
-let e: Int? = nil
-let f: Int? = nil
-
-debugPrint(a ?? nextCounter())      // CHECK-NEXT: 123
-debugPrint(b ?? nextCounter())      // CHECK-NEXT: 0
-debugPrint(c ?? nextCounter())      // CHECK-NEXT: 1
-debugPrint(d ?? nextCounter())      // CHECK-NEXT: 456
-debugPrint(e ?? d ?? nextCounter()) // CHECK-NEXT: 456
-debugPrint(f ?? nextCounter())      // CHECK-NEXT: 2
-
-func nextCounter2() -> Int? {
-  return nextCounter()
-}
+  /*
+  // FIXME: <rdar://problem/17489239> Optional<T>() == nil where T: !Equatable
+  let _: X? = nil
+  let _: X? = X()
 
-debugPrint(c ?? d)                   // CHECK-NEXT: Optional(456)
-debugPrint(c ?? e)                   // CHECK-NEXT: nil
-debugPrint(a ?? nextCounter2())      // CHECK-NEXT: Optional(123)
-debugPrint(b ?? nextCounter2())      // CHECK-NEXT: Optional(3)
-debugPrint(c ?? nextCounter2())      // CHECK-NEXT: Optional(4)
-debugPrint(d ?? nextCounter2())      // CHECK-NEXT: Optional(456)
-debugPrint(e ?? d ?? nextCounter2()) // CHECK-NEXT: Optional(456)
-debugPrint(f ?? nextCounter2())      // CHECK-NEXT: Optional(5)
+  expectFalse(x1 == nil)
+  expectTrue(x1 != nil)
+  expectTrue(x0 == nil)
+  expectFalse(x0 != nil)
 
-import StdlibUnittest
-import Swift
+  expectFalse(nil == x1)
+  expectTrue(nil != x1)
+  expectTrue(nil == x0)
+  expectFalse(nil != x0)
+  */
+}
 
-var OptionalTests = TestSuite("Optional")
+OptionalTests.test("??") {
+  var counter = 0
+  func nextCounter() -> Int { return counter++ }
+  func nextCounter2() -> Int? { return nextCounter() }
+
+  let a: Int? = 123
+  let b: Int? = nil
+  let c: Int? = nil
+  let d: Int? = 456
+  let e: Int? = nil
+  let f: Int? = nil
+
+  expectEqual(a ?? nextCounter(), 123)
```

Comment:
`expectEqual()` accepts the expected value as the first parameter (it matters in the error message when the test fails).